__integrations__/__tests__/system/mongo/mongocdcservice.test.ts

```diff
@@ -0,0 +1,351 @@
+import * as O from "fp-ts/Option";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { Collection, MongoClient } from "mongodb";
+import {
+  disconnectMongo,
+  insertDocument,
+} from "../../../../src/capturer/mongo/utils";
+import {
+  ServiceType,
+  createDatabaseService,
+} from "../../../../src/factory/factory";
+import { MONGO_LEASE_COLLECTION_NAME } from "../../../../src/factory/mongoCDCService";
+import { mongoDBService } from "../../../../src/factory/mongoDBService";
+import { Item } from "../../../../src/factory/service";
+import { MONGODB_CONNECTION_STRING, MONGODB_NAME } from "../../../env";
+import {
+  MONGO_COLLECTION_NAME,
+  createMongoClient,
+  createMongoDBAndCollections,
+  deleteDatabase,
+} from "../../../utils/mongo";
+const service = createDatabaseService(ServiceType.MongoDB);
+var clients: MongoClient[] = [];
+
+beforeAll(async () => {
+  await pipe(
+    createMongoClient(MONGODB_CONNECTION_STRING),
+    TE.chainFirst((client) =>
+      pipe(
+        createMongoDBAndCollections(client, MONGODB_NAME),
+        TE.chain(() => disconnectMongo(client)),
+      ),
+    ),
+
+    TE.getOrElse((e) => {
+      throw Error(
+        `Cannot initialize integration tests - ${JSON.stringify(e.message)}`,
+      );
+    }),
+  )();
+}, 10000);
+
+afterAll(async () => {
+  setTimeout(async () => {
+    for (var client of clients) {
+      await client.close();
+    }
+  }, 15000);
```

Comment:
Unfortunately it's not possible because if the stream is not closed, the client closing will fail. 
So it's necessary to set a timeout and close the client after that the stream is closed.